src/audio/types.h

```diff
@@ -161,11 +161,15 @@ class SampleRate {
         return value();
     }
 
-    static constexpr SampleRate fromDouble(double value) {
-        return SampleRate(static_cast<value_t>(value));
+    static SampleRate fromDouble(double value) {
+        const auto sampleRate = SampleRate(static_cast<value_t>(value));
+        // The sample rate should always be an integer value
+        // and this conversion is supposed to be lossless.
```

Comment:
good idea. This also catches cases where `value` is NaN?

Reply:
Not tested. Probably yes, because there is no NaN integer.